### ISSUE TITLE
Squiz.Arrays.ArrayDeclaration does incorrect align calculation in array with cyrillic keys

### ISSUE THREAD

**Issue by player259** [squizlabs/PHP_CodeSniffer#2144]
Seems that problem with string functions. I didn't found usage of mb_* in project, so all non-latin strings will produce issues.

```php
    public function test()
    {
        // Produces error
        [
            'тип'  => 'авто',
            'цвет' => 'синий',
        ];

        // No error, extra 4 spaces after 'цвет', because it's 4 two-byte character string
        [
            'тип'      => 'авто',
            'цвет'     => 'синий',
        ];
    }
```

```
 15 | ERROR | [x] Array double arrow not aligned correctly; expected 3 space(s) but found -1
    |       |     (Squiz.Arrays.ArrayDeclaration.DoubleArrowNotAligned)
 16 | ERROR | [x] Array double arrow not aligned correctly; expected 1 space(s) but found -3
    |       |     (Squiz.Arrays.ArrayDeclaration.DoubleArrowNotAligned)
```

I've found this substitution list for string functions, if helps
```
'strlen'=>'mb_strlen',
'strpos'=>'mb_strpos',
'substr'=>'mb_substr',
'strtolower'=>'mb_strtolower',
'strtoupper'=>'mb_strtoupper',
'substr_count'=>'mb_substr_count',
'split'=>'mb_split',
'mail'=>'mb_send_mail',
'ereg'=>'mb_ereg',
'eregi'=>'mb_eregi',
'strrchr' => 'mb_strrchr',
'strichr' => 'mb_strichr',
'strchr' => 'mb_strchr',
'strrpos' => 'mb_strrpos',
'strripos' => 'mb_strripos',
'stripos' => 'mb_stripos',
'stristr' => 'mb_stristr'
```

**Comment by gsherwood** [comment]
PHP_CodeSniffer uses the iconv extension as it is enabled by default. This particular sniff was doing it's own strlen calculations instead of using the iconv calculation that is already provided by the core tokenizer, so I've changed the sniff to use those values instead and the bug is now fixed. Thanks for reporting it.
